go_api/dbinit.go

```diff
@@ -0,0 +1,19 @@
+package main
+
+import (
+	"github.com/NUTFes/lottery/go_api/domain"
+
+	"gorm.io/driver/mysql"
+	"gorm.io/gorm"
+)
+
```

Comment:
```suggestion
var (
	db  *gorm.DB
	err error
	dsn = os.Getenv("DSN")
)

func main() {
	dbinit()
}
```
`server.go`15行目の変数定義はここにあったほうがいいですね。
とgoloangは`main`関数が必須なので`main`関数で`dbinit()`を呼ぶ形でいいですね
あと`main`関数を作るとserver.goの`main`関数とダブルので`go_api/cmd/dbinit.go`の中に入れてあげると適切に動くと思われます